This patch fixes the logic that restricts the min and max interval settings.
It makes sure that the internal values are the absolute max and min values.
Patch provided by Chris Carr, see http://bugs.debian.org/468320

Index: ddclient-3.8.0/ddclient
===================================================================
--- ddclient-3.8.0.orig/ddclient	2009-02-18 20:30:17.000000000 +0000
+++ ddclient-3.8.0/ddclient	2009-02-18 20:30:17.000000000 +0000
@@ -1060,7 +1060,7 @@
     $opt{'use'} = 'web' if !define($opt{'use'}) && defined($opt{'web'});
 
     ## sanity check
-    $opt{'max-interval'}       = max(interval(opt('max-interval')), interval(default('max-interval')));
+    $opt{'max-interval'}       = min(interval(opt('max-interval')), interval(default('max-interval')));
     $opt{'min-interval'}       = max(interval(opt('min-interval')), interval(default('min-interval')));
     $opt{'min-error-interval'} = max(interval(opt('min-error-interval')), interval(default('min-error-interval')));
 
